### DISCUSSION TITLE
Question: How can I have different root folder for renderer and pages folder

### DISCUSSION THREAD

**Discussion by kartikkaushal7** [vikejs/vike#1091] [Discussion]
Hello @brillout 
I am creating my own framework based on vite-plugin-ssr.
What I want is that renderer folder should be inside node_modules/framework/renderer, while the pages folder to be at the root of app like

./
- node_modules/framework/renderer
- public //static
- src
- src/App/App.js etc
- src/Components/UIComponents
- pages/ <- This is where I would like him

How can I set something like this ?

**Comment by brillout** [comment]
See https://github.com/brillout/restack and https://github.com/magne4000/solide.

> I am creating my own framework 

What is it you're building?

**Reply by kartikkaushal7** [reply]
I am trying to build my own framework, so I want to hide the renderer code inside node_modules from the users. Moreover, I require that pages folder to be available at the root level.

My question is how can I have renderer and pages folder not in the same root

**Reply by brillout** [reply]
See the examples.
